app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,57 @@
+class PasswordResetsController < ApplicationController
+  before_action :load_user, :valid_user, :check_expiration, only: %i(edit, update)
+
+  def new; end
+
+  def create
+    @user = User.find_by email: params[:password_reset][:email].downcase
+
+    if @user
+      @user.create_reset_digest
+      @user.send_password_reset_email
+      flash[:info] = t "sent_password_reset"
+      redirect_to root_url
+    else
+      flash.now[:danger] = t "email_not_found"
+      render :new
+    end
+  end
+
+  def edit; end
+
+  def update
+    if params[:user][:password].empty?
+      @user.errors.add :password, t("cant_be_empty")
+      render :edit
+    elsif @user.update_attributes user_params
+      log_in @user
+      @user.update_attribute :reset_digest, nil
```

Comment:
convention
```suggestion
      @user.update_attributes reset_digest: nil
```
tuy nhiên việc update đối tượng nên đặt trong model

Reply:
> **tuy nhiên việc update đối tượng nên đặt trong model**

chỗ này em đang không hiểu lắm ạ, chị giải thích thêm cho em với.